### PR TITLE
docs: Correct some mistake in docs.

### PR DESCRIPTION
**Pull request by luyang93**

### CODE REVIEW

File: docs/cluster-workflow-templates.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ argo submit https://raw.githubusercontent.com/argoproj/argo-workflows/master/exa
 The submit a `ClusterWorkflowTemplate` as a `Workflow`:
 
 ```bash
-argo submit --from clusterworkflowtemplate/workflow-template-submittable
+argo submit --from clusterworkflowtemplate/cluster-workflow-template-submittable
```
Comment (agilgur5) [inline docs/cluster-workflow-templates.md:143]: good catch. confirmed this is the name: https://github.com/argoproj/argo-workflows/blob/4aa90111d20ac49d1b1150218bb92b2cd9dcee53/examples/cluster-workflow-template/clustertemplates.yaml#L136

File: docs/workflow-templates.md [hunk 2]
```diff
@@ -344,7 +344,7 @@ argo submit --from workflowtemplate/workflow-template-submittable
 If you need to submit a `WorkflowTemplate` as a `Workflow` with parameters:
 
 ```bash
-argo submit --from workflowtemplate/workflow-template-submittable -p param1=value1
+argo submit --from workflowtemplate/workflow-template-submittable -p message=value1
```
Comment (agilgur5) [inline docs/workflow-templates.md:347]: confirmed: https://github.com/argoproj/argo-workflows/blob/4aa90111d20ac49d1b1150218bb92b2cd9dcee53/examples/workflow-template/templates.yaml#L135

File: docs/async-pattern.md [hunk 3]
```diff
@@ -66,12 +71,12 @@ You may need  an [access token](access-token.md).
 curl --request PUT \
   --url https://localhost:2746/api/v1/workflows/<NAMESPACE>/<WORKFLOWNAME>/resume
   --header 'content-type: application/json' \
-  --header "Authorization: Bearer $ARGO_TOKEN" \
+  --header "Authorization: $ARGO_TOKEN" \
```
Comment (agilgur5) [inline docs/async-pattern.md:74]: I actually fixed these both ~1.5 months ago in #11714. I have [several docs PRs still](https://cloud-native.slack.com/archives/C0510EUH90V/p1696442823801189?thread_ts=1694644337.627309&cid=C0510EUH90V) waiting on review and merge 😕

cc @terrytangyuan

File: docs/async-pattern.md [hunk 4]
```diff
@@ -21,39 +21,44 @@ kind: WorkflowTemplate
 metadata:
   name: external-job-template
 spec:
+  entrypoint: run-external-job
```
Comment (agilgur5) [inline docs/async-pattern.md:24]: this partially replaces #10806, which was abandoned

**Review by agilgur5 — APPROVED** [review]
Confirmed that these are all accurate fixes - thanks!

Note that we may have some merge conflicts with this as there are other, older PRs to `async-pattern.md` that have yet to be reviewed/merged (see in-line)

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**